urllib3/connection.py

```diff
@@ -162,6 +162,37 @@ def connect(self):
         conn = self._new_conn()
         self._prepare_conn(conn)
 
+    def request_chunked(self, method, url, body=None, headers={}):
+        """
+        Alternative to the common request method, which sends the
+        body with chunked encoding and not as one block
+        """
+        header_names = set(k.lower() for k in headers)
+        self.putrequest(method, url, skip_accept_encoding=True)
+        for header, value in headers.items():
+            self.putheader(header, value)
+        if 'transfer-encoding' not in header_names:
+            self.putheader('Transfer-Encoding', 'chunked')
+        self.endheaders()
+
+        if body is not None:
+            if isinstance(body, six.string_types) or isinstance(body, six.binary_type):
+                body = (body,)
+            for chunk in body:
+                if not chunk:
+                    continue
+                len_str = hex(len(chunk))[2:]
+                self.send(len_str.encode('utf-8'))
+                self.send(b'\r\n')
+                if isinstance(chunk, six.binary_type):
+                    self.send(chunk)
+                else:
+                    self.send(chunk.encode('utf-8'))
```

Comment:
Let's save ourselves a line here:

```python
if not isinstance(chunk, six.binary_type):
    chunk = chunk.encode('utf-8')
self.send(chunk)
```

This is a style @shazow slightly prefers. =)